pkg/tsdb/cloudwatch/log_actions.go

```
@@ -0,0 +1,318 @@
+		Custom: map[string]interface{}{
+			"Region": clientRegion,
+		},
+	}
+
+	return dataFrame, nil
+}
+
+func (e *CloudWatchExecutor) executeStopQuery(ctx context.Context, logsClient cloudwatchlogsiface.CloudWatchLogsAPI, parameters *simplejson.Json) (*cloudwatchlogs.StopQueryOutput, error) {
+	queryInput := &cloudwatchlogs.StopQueryInput{
+		QueryId: aws.String(parameters.Get("queryId").MustString()),
+	}
+
+	response, err := logsClient.StopQueryWithContext(ctx, queryInput)
+	if err != nil {
+		awsErr, _ := err.(awserr.Error)
+		if awsErr.Code() == "InvalidParameterException" {
+			response = &cloudwatchlogs.StopQueryOutput{Success: aws.Bool(false)}
+			err = nil
+		} else {
+			err = fmt.Errorf(err.(awserr.Error).Message())
+		}
+	}
+
+	return response, err
+}
+
+func (e *CloudWatchExecutor) handleStopQuery(ctx context.Context, logsClient cloudwatchlogsiface.CloudWatchLogsAPI, parameters *simplejson.Json) (*data.Frame, error) {
+	response, err := e.executeStopQuery(ctx, logsClient, parameters)
+
+	if err != nil {
+		return nil, fmt.Errorf(err.(awserr.Error).Message())
+	}
+
+	dataFrame := data.NewFrame("StopQueryResponse", data.NewField("success", nil, []bool{*response.Success}))
+	return dataFrame, nil
+}
+
+func (e *CloudWatchExecutor) executeGetQueryResults(ctx context.Context, logsClient cloudwatchlogsiface.CloudWatchLogsAPI, parameters *simplejson.Json) (*cloudwatchlogs.GetQueryResultsOutput, error) {
+	queryInput := &cloudwatchlogs.GetQueryResultsInput{
+		QueryId: aws.String(parameters.Get("queryId").MustString()),
+	}
+
+	return logsClient.GetQueryResultsWithContext(ctx, queryInput)
+}
+
+func (e *CloudWatchExecutor) handleGetQueryResults(ctx context.Context, logsClient cloudwatchlogsiface.CloudWatchLogsAPI, parameters *simplejson.Json, refID string) (*data.Frame, error) {
+	getQueryResultsOutput, err := e.executeGetQueryResults(ctx, logsClient, parameters)
+	if err != nil {
+		return nil, fmt.Errorf(err.(awserr.Error).Message())
```

Why not return the original error?